### PR TITLE
Implement page-message allowlist to improve safe news-related urls drop rate

### PR DESCRIPTION
**Pull request by LorenzoMinto**
This PR implements a page-message allowlist and prevents the dropping of news-related URLs like:
https://www.npr.org/2022/10/18/1129521692/matthew-deperno-dana-nessel-michigan-attorney-general-voting-machines
https://www.wsj.com/articles/china-economy-xi-jinping-ideology-11666016306
https://www.propublica.org/article/wisconsin-election-midterms-voting-2022#1460399
https://www.huffpost.com/entry/singer-mikaben-dies-after-paris-collapse_n_634d76a8e4b051268c4c5f84

### CODE REVIEW

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:1649]: Would it be safer to include a prefix match including protocol here? e.g. `https://play.google.com/store/apps/details?id=...`

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:3368]: Nit.
```suggestion
          const whitelisted = WebDiscoveryProject.whitelist.some(whitelist_regex => whitelist_regex.test(activeURL));
```

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:1645]: Nit. Maybe better to call this one `allowlist`.

**Comment by remusao** [comment]
I'll have to check what's going on with CI.

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:1649]: Nit.
```suggestion
    /^https:\/\/play\.google\.com\/store\/apps\/details\?id=[^\/]+$/,
    /^https:\/\/apps\.apple\.com\/\D{2}\/app\/[\w-]+\/id\d+$/,
    /^https:\/\/www\.npr\.org\/\d{4}\/\d{2}\/\d{2}/,
    /^https:\/\/variety\.com\/\d{4}\//,
```

**Review by remusao — COMMENTED** [review]
This makes sense to me. Would it be possible to re-use the script you created initial to label URLs to also act as a regression to test to check which URLs are classified differently after this change?

**Comment by LorenzoMinto** [comment]
I've added a table on the minispec specifying what URLs are affected by each PR. For this I've limited it to `apps.apple.com` and `play.google.com` as they are two of the most impacted URLs and share a quick fix.

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (fmarier) [inline modules/web-discovery-project/sources/web-discovery-project.es:1868]: I'd suggest adding a comment here to say that the value is allowlisted since the identifier `alw` may not be obvious to the reader.

**Comment by LorenzoMinto** [comment]
@remusao @fmarier I've updated the PR to focus on news-related URLs for which we have built some Utility Regression tests.

8 out of the 38 sources that we serve in Brave News had URL patterns that were being erroneously blocked by WDP (some examples of such URLs can be found in the PR description above). The URLs have been sampled 1 per source from the article URLs served by Brave News.

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:36]: Something to remove before merging?

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:1950]: Nit. No need to concatenate if `log_` is not doing anything (log disabled). Also, even though JavaScript will not complain about it, `query_string` can be null and it sounds wrong to concatenate `null` to a string. 
```suggestion
            _log('DLU failed: long number in the query string: ' + url_parts.query_string);
```

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:1956]: Nit. Same as above.
```suggestion
          _log('DLU failed: long number in path: ', url_parts.path);
```

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:1973]: Nit. Same as above.
```suggestion
            _log('DLU failed: hash in the URL ', vpath[i]);
```

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:1986]: Nit. Same as above.
```suggestion
            _log('DLU failed: hash in the path ', cstr);
```

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:2053]: Nit. Same as above.
```suggestion
              _log("Prohibited keyword found: ", url_parts.query_string);
```

File: modules/web-discovery-project/sources/web-discovery-project.es
Comment (remusao) [inline modules/web-discovery-project/sources/web-discovery-project.es:2061]: Nit. Same as above.
```suggestion
              _log("Prohibited keyword found: ", path_query_string);
```

**Review by remusao — APPROVED** [review]

**Review by fmarier — APPROVED** [review]

**Pull request merged**
